server/middlewares/stats.js

```diff
@@ -41,44 +41,81 @@ const parseTxBodyOutAmount = (txBody) => {
   }
 }
 
-const incrCountersBy = (key, value) => {
-  const [year, month, day] = getSlicedDate()
-  client.incrby(`${key}:total`, value)
-  client.incrby(`${key}:monthly:${year}-${month}`, value)
-  client.incrby(`${key}:daily:${year}-${month}-${day}`, value)
+const initVisitor = (trackingID) => {
+  return ua({
+    tid: trackingID,
+    cid: 555, //google analytics anonymous visitor
+    strictCidFormat: false, //to accept cid integer instead of UUID v4 format
+  })
+}
+
+const trackPageView = (...args) => {
+  const visitor = initVisitor(backendConfig.ADALITE_TRACKING_ID)
+  visitor.pageview(...args).send()
+}
+
+const trackEvent = (...args) => {
+  const visitor = initVisitor(backendConfig.ADALITE_TRACKING_ID)
+  visitor.event(...args).send()
 }
 
 const trackVisits = (req, res, next) => {
+  if (!backendConfig.ADALITE_TRACKING_ID) {
+    next()
+    return
+  }
+
   // 'x-forwarded-for' due to internal heroku routing
   const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress
-
   const mydevice = device(req.headers['user-agent'])
 
   if (!knownIps.has(ip) && !mydevice.is('bot')) {
-    knownIps.add(ip)
-    incrCountersBy('visits', 1)
+    try {
+      trackPageView('/')
+      knownIps.add(ip)
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(`Tracking page view failed - ${err.message}`)
+    }
   }
 
   next()
 }
 
 const trackTxSubmissions = mung.json((body, req, res) => {
-  if (req.originalUrl === '/api/txs/submit' && req.method === 'POST') {
-    const txSubmissionType =
-      tokenMatches(req.get('token')) &&
-      isSameOrigin(req.get('origin'), process.env.ADALITE_SERVER_URL)
-        ? 'txSubmissions'
-        : 'otherTxSubmissions'
-    const txSubmissionSuccess = body.Right ? 'successful' : 'unsuccessful'
-
-    incrCountersBy(`${txSubmissionType}:${txSubmissionSuccess}`, 1)
-
-    if (txSubmissionSuccess === 'successful') {
-      const {txBody} = req.body
-      const txOutAmount = parseTxBodyOutAmount(txBody)
+  if (!backendConfig.ADALITE_TRACKING_ID) {
+    return
+  }
 
-      incrCountersBy(`${txSubmissionType}:sentOut`, txOutAmount)
+  try {
```

Comment:
the try-catch imho belongs inside the `if` block, I would wrap only the `if (txSubmissionSuccess === 'successful') {` block with it.